src/ddo/interfaces/Service.ts

```diff
@@ -1,12 +1,27 @@
 import { Metadata } from './Metadata'
 import { Status } from './Status'
 
-export type ServiceType = 'authorization' | 'metadata' | 'access' | 'compute'
+export interface CustomData {
+  name: string
+  type: string
+  label: string
+  required: boolean
+  options?: any
+  description: string
+}
 
+export interface RequiredData {
+  userdata?: CustomData[]
```

Comment:
camelCase consistency, so should be `userData` & `algoData`

Reply:
also `data` is very generic in multiple places in this PR, can we come up with something more precise. Otherwise we end calling everything `data` even if we can be more precise like `parameters`, `options`, `metadata`, `user input` and so on

Reply:
`userCustomParameters` would be more precise i think, since this is what their purpose is.